iceoryx_posh/test/moduletests/test_shared_memory_user.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2021 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_posh/internal/roudi_environment/roudi_environment.hpp"
+#include "iceoryx_posh/internal/runtime/shared_memory_user.hpp"
+
+#include "test.hpp"
+
+using namespace ::testing;
+using namespace iox::runtime;
+using namespace iox::roudi;
+
+namespace iox
+{
+namespace test
+{
+class SharedMemoryUserTest : public Test
+{
+  public:
+    RouDiEnvironment m_roudiEnv{iox::RouDiConfig_t().setDefaults()};
+    const uint64_t segmentId{1U};
+    rp::BaseRelativePointer::offset_t segmentManagerAddressOffset{0U};
+    void SetUp(){}
+    void TearDown(){}
+};
+
+TEST_F(SharedMemoryUserTest, ConstructorShmObjectWithZeroTopicSizeAndDoMapSharedMemoryIntoThreadBoolSetToTrueReturnError)
```

Comment:
This title describes the implementation and not what we actually do and expect. Maybe a title like `CreateAndMapZeroSizedSharedMemoryObjectFails` describes better what is actually done from a technical point of view.

Please adapt all the other titles as well.